### PR TITLE
UI-2108: Extend Group Marker for Individual Pin Highlight

### PR DESCRIPTION
**Pull request by mallik-svmx**
Application controls following settings with click event Listener to get the desired behavior:
- Extend Group Marker for Individual Pin Highlight
- Add lat/lng, and zoom prop to px-map component
- Add lat/lng, and opened prop to px-map-marker-group component
- marker-group-component navigates to selected marker, if  a valid one exists

### CODE REVIEW

File: px-map-behavior-marker-group.es6.js [hunk 1]
```diff
@@ -270,6 +318,30 @@
...
+      // automatically open popup if marker-group element has opened prop
+      if (lastOptions.opened !== nextOptions.opened) {
+        if (nextOptions.opened && !this.elementInst.isPopupOpen()) {
+          const nextOptionsData = nextOptions.data;
+          if (nextOptionsData && nextOptionsData.features.length) {
+            // find if selected item exists
+            const selectedFeature = nextOptionsData.features.find(feature => {
+              const geoSelected = [nextOptions.lng, nextOptions.lat];
+              const geoToBeSelected = feature.geometry.coordinates;
+              return (geoSelected[0] === geoToBeSelected[0] && geoSelected[1] === geoToBeSelected[1]);
```
Comment (leslieli) [inline px-map-behavior-marker-group.es6.js]: Will it open all the popup for all the items which have the same location/geocode in marker-group?

File: px-map-behavior-root.es6.js [hunk 2]
```diff
@@ -513,7 +513,9 @@
 
     createInst(options) {
       const mapEl = Polymer.dom(this.root).querySelector('#map');
-      const mapInst = L.map(mapEl, options);
+      // need mapInst to automatically open popup in marker-group component
+      // L.popup().setLatLng([latlng]).setContent(popupContent).openOn(mapInst)
+      const mapInst = window.mapInst = L.map(mapEl, options);
```
Comment (mallik-svmx) [inline px-map-behavior-root.es6.js]: I don't like globals neither, created [UI-2432](https://servicemax.atlassian.net/browse/UI-2432) as tech debt.

**Comment by mallik-svmx** [comment]
@joshsylvester @leslieli @hanzge @sheela-svmx - will resolve the conflict otherwise, its ready for review.

File: px-map-behavior-root.es6.js (hunk 2 above)
Comment (mallik-svmx) [reply px-map-behavior-root.es6.js]: Removed global.

File: examples/marker-clusters-with-popups.html [hunk 3]
```diff
@@ -36,22 +36,43 @@
...
+      padding: 5px 0;
+    }
   </style>
 </head>
 <body>
   <section class="wrap">
     <dom-bind>
       <template is="dom-bind">
-        <px-map fit-to-markers flex-to-size on-px-map-moved="mapMove">
+        <px-map id="pxMap" flex-to-size on-px-map-moved="mapMove">
```
Comment (sheela-svmx) [inline examples/marker-clusters-with-popups.html]: I am curious why fit-to-markers was removed?

File: examples/marker-clusters-with-popups.html [hunk 4]
```diff
@@ -683,6 +704,36 @@
...
+      // pass-in lat/lng, opened prop to px-marker-group to compare and select feature clicked
+      // track if popup opened
+      var pxMap = document.getElementById('pxMap');
+      var pxMarkerGroup = document.getElementById('pxMarkerGroup');
+      document.getElementById('markerOne').addEventListener('click', function() {
+        const [LNG, LAT] = featCollection.features[0].geometry.coordinates;
+        const { id } = featCollection.features[0];
+        pxMap.lat = LAT;
+        pxMap.lng = LNG;
+        pxMap.zoom = 17; //single marker centered to map
```
Comment (sheela-svmx) [inline examples/marker-clusters-with-popups.html]: should this zoom value be customizable?

Comment (mallik-svmx) [reply examples/marker-clusters-with-popups.html]: Can you please elaborate? what do you mean by customizable?

Comment (mallik-svmx) [reply examples/marker-clusters-with-popups.html]: Yes. It is a hardcoded arbitrary zoom value in here that(application passed-in value) opens popup for item clicked "exactly" positioning to the marker including marker in cluster.

File: examples/marker-clusters-with-popups.html (hunk 3 above)
Comment (leslieli) [reply examples/marker-clusters-with-popups.html]: OK, thanks.

**Pull request merged**